File: gerenciadorDeTurma/eventBus.ts
type EventHandler = (payload: any) => void;

class EventBus {
  private handlers = new Map<string, EventHandler[]>();

  subscribe(event: string, handler: EventHandler) {
    if (!this.handlers.has(event)) {
      this.handlers.set(event, []);
    }

    this.handlers.get(event)?.push(handler);
  }

  publish(event: string, payload: any) {
    const handlers = this.handlers.get(event);

    if (handlers) {
      for (const handler of handlers) {
        handler(payload);
      }
    }
  }
}

export const eventBus = new EventBus();

File: gerenciadorDeTurma/enrollment/shared/event_bus.py
class Events ():
    def __init__(self):    
        self.classRoomCreated= "ClassRoomCreated"
        self.studentEnrolled= "studentEnrolled"    
        self.studentRegistered= "studentRegistered"
        self.studentNotified= "studentNotified" 


class EventBus:
    def __init__(self):
        self.handlers = {}

    def subscribe(self, event, handler):
        if event not in self.handlers:
            self.handlers[event] = []
        self.handlers[event].append(handler)

    def publish(self, event, payload):
        if event in self.handlers:
            for handler in self.handlers[event]:
                handler(payload)

event_bus = EventBus()
events = Events()
